usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -274,13 +274,16 @@ def fetch_award_details(filter_q, mapper_fields):
     return Award.objects.filter(**filter_q).values(*vals).annotate(**ann).first()
 
 
-def fetch_parent_award_from_piid(piid):
-    parent_award_id = (
-        ParentAward.objects.filter(generated_unique_award_id__icontains=piid)
-        .values("generated_unique_award_id")
-        .first()
-    )
-    return parent_award_id
+def fetch_parent_award_from_piid_agency(piid, fpds_agency):
+    if piid:
+        parent_unique_key = "CONT_IDV_{}_{}".format(piid, fpds_agency)
+        parent = (
+            ParentAward.objects.filter(generated_unique_award_id=parent_unique_key)
+            .values("generated_unique_award_id")
+            .first()["generated_unique_award_id"]
```

Comment:
[.first()](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.first) will either return the first record found or `null`. If no parent is found this code will cause an exception since it assumes there will be a dict to lookup `generated_unique_award_id`